zoom_rooms/indico_zoom_rooms/tasks.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2024 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+from pprint import pformat
+
+import requests
+from celery.schedules import crontab
+from celery.utils.log import get_task_logger
+
+from indico.core.celery import celery
+from indico.core.db import db
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.models.events import Event
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.util.string import strip_control_chars
+
+from indico_zoom_rooms.models import ZoomRoomsAction, ZoomRoomsQueueEntry
+
+
+def obj_from_locator(locator: dict) -> Contribution | SessionBlock | Event:
+    """Convert a locator dict into a usable DB object."""
+    if 'contrib_id' in locator:
+        return Contribution.get(locator['contrib_id'])
+    elif 'block_id' in locator:
+        return SessionBlock.get(locator['block_id'])
+    elif 'event_id' in locator:
+        return Event.get(locator['event_id'])
+    else:
+        raise ValueError("Can't decode locator")
+
+
+logger = get_task_logger(__name__)
+
+
+def _send_request(method: str, zr_id: str, entry_id: str, data: dict | None = None) -> bool:
+    """Send a request to the corresponding HTTP service."""
+    from indico_zoom_rooms.plugin import ZoomRoomsPlugin
+
+    path = f'/api/v1/users/{zr_id}/events/{entry_id}'
+    if not (url := ZoomRoomsPlugin.settings.get('service_url')):
+        raise KeyError('service_url is not set!')
+    url = url.rstrip('/') + path
+
+    if not (token := ZoomRoomsPlugin.settings.get('token')):
+        raise KeyError('token is not set!')
+    try:
+        res = requests.request(
+            method,
+            url,
+            json=data,
+            headers={'Authorization': f'Bearer {token}'},
+            timeout=ZoomRoomsPlugin.settings.get('timeout'),
+        )
+    except requests.Timeout:
+        logger.warning('Request timed out')
+    except requests.RequestException:
+        logger.exception('%s request failed:\nURL: %s\nData: %s', method, url, pformat(data))
+    else:
+        logger.info('%s request to %s finished with status %r and body %r', method, path, res.status_code, res.text)
+        if res.ok:
+            return True
```

Comment:
You could do `res.raise_for_status()` in the `try` block and then catch `requests.HTTPError` to handle non-successful status codes